src/Renode/Network/NetworkServer/IServerModule.cs

```diff
@@ -8,14 +8,22 @@
 using System;
 using System.Net;
 using System.Collections.Generic;
-
 using PacketDotNet;
 
 namespace Antmicro.Renode.Network
 {
     public interface IServerModule
     {
-        void HandleUdp(IPEndPoint source, UdpPacket packet, Action<IPEndPoint, UdpPacket> callback);
     }
-}
 
+    public interface IUdpServerModule : IServerModule
+    {
+        void HandleUdp(IPEndPoint source, UdpPacket packet,
+            Action<IPEndPoint, UdpPacket> callback);
+    }
+
+    public interface IIcmpServerModule : IServerModule
+    {
+        void HandleIcmp(IPv4Packet packet);
```

Comment:
I would consider having a similar API to UDP - with source, with packet, with callback